app/models/health/qualifying_activity.rb

```diff
@@ -510,21 +570,50 @@ def any_submitted_of_type_for_day_for_patient?
       same_of_type_for_day_for_patient.submitted.exists?
     end
 
-    def occurred_after_enrollment?
-      date_of_activity.present? && patient&.patient_referral.present? && date_of_activity >= patient.patient_referral.enrollment_start_date
+    def occurred_during_any_enrollment?
+      date_of_activity.present? && patient.patient_referrals.active_within_range(start_date: date_of_activity, end_date: date_of_activity).exists?
     end
 
-    def occurred_within_three_months_of_enrollment
-      date_of_activity.present? && patient&.patient_referral.present? && patient&.enrollment_start_date.present? && date_of_activity <= (patient.outreach_cutoff_date)
+    def occurred_during_enrollment?
+      date_of_activity.present? && patient.patient_referrals.contributing.active_within_range(start_date: date_of_activity, end_date: date_of_activity).exists?
+    end
+
+    def occurred_within_three_months_of_enrollment?
+      date_of_activity.present? && patient.current_days_enrolled <= 90
+    end
+
+    def patient_has_valid_care_plan?
+      return false unless patient.care_plan_renewal_date.present?
+      return false unless date_of_activity.present?
+
+      date_of_activity >= patient.care_plan_provider_signed_date && date_of_activity < patient.care_plan_renewal_date
+    end
+
+    def in_care_plan_development_period?
+      date_of_activity.present? && patient.current_days_enrolled <= 150
+    end
+
+    def in_validity_window?
+      return false unless date_of_activity.present?
+
+      if outreach?
+        return false unless occurred_within_three_months_of_enrollment?
+      end
```

Comment:
These nested ifs with trailing unlesses are hard for me to read.  Could this not be written as `return false if outreach? && ! occurred_within_three_months_of_enrollment?`?  Or, maybe we want a method of `outside_3_month_window?` so it wouldn't need the negation.